### PR TITLE
benchmark contract execution

### PR DESCRIPTION
**Pull request by yihuang**
Closes: #431

## Description

```
$ go test ./x/evm/keeper -v -run="^$" -bench=".*"
BenchmarkTokenTransfer
BenchmarkTokenTransfer-16    	    2996	    347253 ns/op	  119528 B/op	    1756 allocs/op
BenchmarkEmitLogs
BenchmarkEmitLogs-16         	       1	1071863815 ns/op	1515836464 B/op	13601697 allocs/op
```

it seems we do have some significant performance issue with log emit. takes a second to emit 1000 logs.

______

For contributor use:

- [x] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added  relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [x] Re-reviewed `Files changed` in the Github PR explorer

______

For admin use:

- [ ] Added appropriate labels to PR (ex. `WIP`, `R4R`, `docs`, etc)
- [ ] Reviewers assigned
- [ ] Squashed all commits, uses message "Merge pull request #XYZ: [title]" ([coding standards](https://github.com/tendermint/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

**Comment by leejw51crypto** [comment]
how about the speed without logs?

**Comment by fedekunze** [comment]
Seems that some tests are failing

File: x/evm/keeper/keeper_test.go [hunk 1]
```diff
@@ -146,6 +183,46 @@ func (suite *KeeperTestSuite) initKeepersWithmAccPerms() (authkeeper.AccountKeep
 	return authKeeper, keeper
 }
 
+// DeployTestContract deploy a test erc20 contract and returns the contract address
+func (suite *KeeperTestSuite) DeployTestContract(t require.TestingT, owner common.Address, supply *big.Int) common.Address {
```
[the remark this answers is not in the document]
Comment (yihuang) [reply x/evm/keeper/keeper_test.go:187]: To make it work on both `testing.B` and `testing.T`.

File: x/evm/keeper/benchevm_test.go [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+	msg := txBuilder(suite, contractAddr)
+	msg.From = suite.address.Hex()
+	err := msg.Sign(ethtypes.LatestSignerForChainID(suite.app.EvmKeeper.ChainID()), suite.signer)
+	require.NoError(b, err)
+
+	b.ResetTimer()
+	b.StartTimer()
+	for i := 0; i < b.N; i++ {
+		ctx, _ := suite.ctx.CacheContext()
```
[the remark this answers is not in the document]
Comment (yihuang) [reply x/evm/keeper/benchevm_test.go:47]: To make each run independent, we don't mutate `suite.ctx`.

File: x/evm/keeper/keeper_test.go [hunk 3]
```diff
@@ -50,20 +80,22 @@ type KeeperTestSuite struct {
 
 	appCodec codec.Codec
 	signer   keyring.Signer
+
+	EvmDenom string
```
[the remark this answers is not in the document]
Comment (yihuang) [reply x/evm/keeper/keeper_test.go]: removed now, changed to a method.

File: x/evm/keeper/keeper_test.go [hunk 4]
```diff
@@ -50,20 +80,22 @@ type KeeperTestSuite struct {
...
 }
 
-func (suite *KeeperTestSuite) SetupTest() {
+func (suite *KeeperTestSuite) DoSetupTest(t require.TestingT) {
```
[the remark this answers is not in the document]
Comment (yihuang) [reply x/evm/keeper/keeper_test.go:87]: To make it work on both testing.B and testing.T.

Added a line of comment to explain.

**Pull request merged**